lib/vagrant/pre-rubygems.rb

```diff
@@ -1,34 +0,0 @@
-# This file is to be loaded _before_ any RubyGems are loaded. This file
-# initializes the Bundler context so that Vagrant and its associated plugins
-# can load properly, and then execs out into Vagrant again.
-
-require_relative "shared_helpers"
-
-if defined?(Bundler)
-  require "bundler/shared_helpers"
-  if Bundler::SharedHelpers.in_bundle? && !Vagrant.very_quiet?
-    puts "Vagrant appears to be running in a Bundler environment. Your "
-    puts "existing Gemfile will be used. Vagrant will not auto-load any plugins"
-    puts "installed with `vagrant plugin`. Vagrant will autoload any plugins in"
-    puts "the 'plugins' group in your Gemfile. You can force Vagrant to take over"
-    puts "with VAGRANT_FORCE_BUNDLER."
-    puts
```

Comment:
I bring my two cents here, after I've just noticed that the plugins auto-loading is no longer disabled when running Vagrant from a Bundler environment. This is "breaking" my Vagrant installation from source, as Vagrant now tries to load all the plugins defined in `~/.vagrant.d/plugins.json`, but fails to find them in such context.

Thanks to the new `VAGRANT_DISABLE_PLUGIN_INIT` environment variable, I can force to disable the plugins initialisation, which is fair enough for working on the Vagrant project itself.

@chrisroberts In order to avoid misunderstanding situations for people wanting to contribute to the Vagrant project, I guess it would probably be worth to

* consider disabling the plugin auto-loading _by default_ when running in a "development" context (or be sure to not conflict with plugins installed in an official Vagrant setup).
* update the related [README.md](https://github.com/mitchellh/vagrant/blob/master/README.md#contributing-to-vagrant) section, especially the note copied below

> **NOTE:** By default running Vagrant via `bundle` will disable plugins.
This is necessary because Vagrant creates its own private Bundler context
(it does not respect your Gemfile), because it uses Bundler to manage plugin
dependencies.

What do you think? Sorry if I missed something obvious, as I haven't taken time to fully understand the changes brought here. I merely wanted to report a "contributor experience" feedback. Hope it helps and makes sense for you...